source/Set.h

```diff
@@ -36,6 +36,8 @@ class Set {
 	
 	typename std::map<std::string, Type>::iterator begin() { return data.begin(); }
 	typename std::map<std::string, Type>::const_iterator begin() const { return data.begin(); }
+	typename std::map<std::string, Type>::iterator find(std::string &key) { return data.find(key); }
```

Comment:
Not sure if should be done in this PR but since we're repeating the types here a lot we should create type alias. Either just for the map or also for the iterators.

Also the parameter should be taken as `const&`.